src/snowflake/connector/log_configuration.py

```diff
@@ -0,0 +1,113 @@
+
+    def parse_config_file(self):
+        config_file_path = self.search_config_file()
+        if config_file_path:
+            with open(config_file_path) as f:
+                data = json.load(f)
+                if (
+                    "common" not in data
+                    or "log_level" not in data["common"]
+                    or "log_path" not in data["common"]
+                ):
+                    raise ValueError(
+                        f"config file at {config_file_path} is not in correct form, please verify your config file"
+                    )
+                data = data["common"]
+                self.log_level = data["log_level"]
+                self.log_path = data["log_path"]
+                if not os.access(self.log_path, os.R_OK | os.W_OK):
+                    raise PermissionError(
+                        f"log path: {self.log_path} is not accessible, please verify your config file"
+                    )
+                if self.log_level not in self.log_levels:
+                    raise ValueError(
+                        f"given log level: {self.log_level} is not valid, allowed log levels: {', '.join(list(self.log_levels.keys()))}"
+                    )
+
+    def search_config_file(self) -> str:
+        # check connection parameter
+        if self.CLIENT_CONFIG_FILE:
+            if is_full_path(self.CLIENT_CONFIG_FILE):
+                return self.CLIENT_CONFIG_FILE
+            else:
+                raise FileNotFoundError(
+                    f"given file path {self.CLIENT_CONFIG_FILE} is not full path"
+                )
+
+        # check environment parameter
+        if self.SF_CLIENT_CONFIG_FILE:
+            if is_full_path(self.SF_CLIENT_CONFIG_FILE):
+                return self.SF_CLIENT_CONFIG_FILE
+            else:
+                raise FileNotFoundError(
+                    f"given file path {self.SF_CLIENT_CONFIG_FILE} is not full path"
+                )
+
+        # search under driver directory
+        drive_directory = os.path.dirname(inspect.getfile(SnowflakeConnection))
+        files = os.listdir(drive_directory)
+        if self.config_file_name in files:
+            return os.path.join(drive_directory, self.config_file_name)
+
+        # search in user home directory
+        home_dir = os.path.expanduser("~")
+        files = os.listdir(home_dir)
+        if self.config_file_name in files:
+            return os.path.join(home_dir, self.config_file_name)
+
+        # return empty str if file is not found
+        return ""
+
+    def create_log(self):
+        logging.basicConfig(
+            filename=self.log_path, level=logging.getLevelName(self.log_level)
+        )
+        for logger_name in ["snowflake.connector", "botocore", "boto3"]:
+            logger = logging.getLogger(logger_name)
+            logger.setLevel(logging.getLevelName(self.log_level))
+            ch = logging.FileHandler(os.path.join(self.log_path, self.log_file_name))
```

Comment:
I think if we use the TOML file approach, the `[logging]path` value should be the full path to the log file, and it must be an absolute path.  If we allow for relative paths, then we have to think about where that path is relative to, and I'd rather not guess that, or put it in the driver directory, etc.  I suggest a default path of `$SNOWFLAKE_HOME/logs` and any relative path would be relative to that default.  (`$SNOWFLAKE_HOME` of course defaults to `~/.snowflake/`.